components/brave_wallet_ui/panel/async/wallet_panel_async_handler.ts

```diff
@@ -170,26 +172,27 @@ handler.on(PanelActions.cancelConnectToSite.getType(), async (store: Store, payl
 })
 
 handler.on(PanelActions.cancelConnectHardwareWallet.getType(), async (store: Store, txInfo: TransactionInfo) => {
-  const hardwareAccount = await findHardwareAccountInfo(txInfo.fromAddress)
-  if (!hardwareAccount || !hardwareAccount.hardware) {
+  const found = await findHardwareAccountInfo(txInfo.fromAddress)
+  if (!found || !found.hardware) {
     return
   }
-  const apiProxy = getWalletPanelApiProxy()
-  await cancelHardwareOperation(apiProxy, hardwareAccount.hardware.vendor)
+  const info: HardwareInfo = found.hardware
+  await cancelHardwareOperation(info.vendor as HardwareVendor)
   // Navigating to main panel view will unmount ConnectHardwareWalletPanel
   // and therefore forfeit connecting to the hardware wallet.
   await store.dispatch(PanelActions.navigateToMain())
 })
 
 handler.on(PanelActions.approveHardwareTransaction.getType(), async (store: Store, txInfo: TransactionInfo) => {
-  const hardwareAccount = await findHardwareAccountInfo(txInfo.fromAddress)
-  if (!hardwareAccount || !hardwareAccount.hardware) {
+  const found = await findHardwareAccountInfo(txInfo.fromAddress)
+  if (!found || !found.hardware) {
```

Comment:
ditto

Reply:
Here I think we should always have hardware transaction and never fall into this condition but if it happend user could be confused, @onyb any thoughts on it? mind to add a new state/message to ConnectHardwareView for this?